includes/actions/update-next-payment.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace AutomateWoo_Subscriptions;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Change a subscription's currency.
+ *
+ * While the currency is not a line item, this class still extends Action_Subscription_Edit_Item_Abstract
+ * as it provides many useful methods for editing a subscription's currency.
+ *
+ * @class Action_Subscription_Update_Next_Payment
+ * @since 1.0.0
+ */
+class Action_Subscription_Update_Next_Payment extends \AutomateWoo\Action {
```

Comment:
I think instead of directly extending `Action` we can create an abstract class called `Action_Subscription`.

The following things will then get moved to the abstract class from here:

```php
/**
 * A subscription is needed so that it can be edited by instances of this action.
 *
 * @var array
 */
public $required_data_items = [ 'subscription' ];
```

```php
$this->group  = __( 'Subscription', 'automatewoo' );
```

```php
/**
 * Get the subscription passed in by the workflow's trigger.
 *
 * @return \WC_Subscription|false
 */
protected function get_subscription_to_edit() {
	return $this->workflow->data_layer()->get_subscription();
}
```